src/run/run.c

```diff
@@ -1713,7 +1713,11 @@ static int start_transient_service(sd_bus *bus) {
                 } else if (arg_transport == BUS_TRANSPORT_MACHINE) {
                         _cleanup_(sd_bus_unrefp) sd_bus *system_bus = NULL;
                         _cleanup_(sd_bus_message_unrefp) sd_bus_message *pty_reply = NULL;
-                        const char *s;
+                        const char *machine_name, *p;
+
+                        r = parse_machine_spec(arg_host, &machine_name, /* ret_user = */ NULL);
```

Comment:
hmm, just ignoring the user seems weird to me. We either should do something useful with it or return an error.

Reply:
I think we need a `OpenMachinePTYEx` method to allow chowning the newly created pty slave to the target user?